### PR TITLE
gh-106293: Fix typos in Objects/object_layout.md

### LINKED ISSUE #106293: Typos, Grammar mistakes in object_layout.md
Found multiple typos, and grammar mistakes which can be improved in `Objects/object_layout.md`

### Linked PRs
* gh-106294
* gh-114158

### PR DESCRIPTION
**Pull request by manosriram**
Fixes #106293

* Issue: gh-106293

### CODE REVIEW

File: Objects/object_layout.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ the values pointer, to enable the (legacy) C-API function
 * ob_refcnt
 * ob_type
 
-For a "normal" Python object, that is one that doesn't inherit from a builtin
+For a "normal" Python object, i.e., the one that doesn't inherit from a builtin
```
Comment (terryjreedy) [inline Objects/object_layout.md]: 'the one' is not correct, and a bit worse than previously.  Following is better.
```suggestion
For a "normal" Python object, one that doesn't inherit from a builtin
```

File: Objects/object_layout.md [hunk 2]
```diff
@@ -7,14 +7,14 @@ Each Python object starts with two fields:
...
-which the form the header common to all Python objects, for all versions,
+which form the header common to all Python objects, for all versions,
 and hold the reference count and class of the object, respectively.
 
 ## Pre-header
 
 Since the introduction of the cycle GC, there has also been a pre-header.
 Before 3.11, this pre-header was two words in size.
-It should be considered opaque to all code except the cycle GC.
+It should be considered opaque to all of the code except the cycle GC.
```
Comment (terryjreedy) [inline Objects/object_layout.md]: This was ok and I do not see this as improvement, though someone else might disagree.
```suggestion
It should be considered opaque to all code except the cycle GC.
```

**Review by terryjreedy — CHANGES_REQUESTED** [review]
My score: 2 correct, 1 bad fix for a real problem, 1 maybe not needed.

@CAM-Gerlach What do you think?  Feel free to merge if you think appropriate.

The blurb was not really needed for this, but the bot cannot tell.

File: Objects/object_layout.md (hunk 1 above)
Comment (manosriram) [reply Objects/object_layout.md]: Yeah, that sounds much better. Updated.

**Comment by manosriram** [comment]
I have made the requested changes; please review again

File: Misc/NEWS.d/next/Documentation/2023-06-30-21-40-28.gh-issue-106293.z4sD7h.rst
Comment (methane) [inline Misc/NEWS.d/next/Documentation/2023-06-30-21-40-28.gh-issue-106293.z4sD7h.rst:1]: news is not needed for this type of PR.
changelog is for users. And this change is not user visible.

Comment (manosriram) [reply Misc/NEWS.d/next/Documentation/2023-06-30-21-40-28.gh-issue-106293.z4sD7h.rst:1]: Removed the commit.

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
@terryjreedy:

> Feel free to merge if you think appropriate.

I applied your suggestion. Enabling auto-merge.

**Pull request merged**